addons/website/static/src/scss/website.scss

```diff
@@ -1014,22 +1014,35 @@ header {
                 }
             }
 
-            .o_connected_user:not(.editor_has_snippets) header & {
+            .o_connected_user:not(.editor_has_snippets):not(.o_fullscreen) header & {
                 // TODO this has to be reviewed
                 $-header-template: o-website-value('header-template');
                 @if not index(('Contact', 'boxed', 'magazine'), $-header-template) {
                     top: -$o-navbar-height;
                     padding-top: $o-navbar-height;
                 } @else if $-header-template == 'magazine' {
                     @include media-breakpoint-up(lg) {
-                        position: fixed;
+                        padding-top: $o-navbar-height;
+                    }
+                }
+            }
 
-                        #top_menu {
-                            align-items: normal !important;
-                        }
+            .editor_has_snippets:not(.o_fullscreen) header & {
+                @if (o-website-value('header-template') == 'magazine') {
+                    @include media-breakpoint-up(lg) {
+                        right: $o-we-sidebar-width;
                     }
                 }
             }
+
+            @if o-website-value('header-template') == 'magazine' {
+                @include media-breakpoint-up(lg) {
+                    position: fixed;
+                }
+                #top_menu {
+                    align-items: normal !important;
+                }
+            }
```

Comment:
> I take care of it in this PR, if you agree

But needs to be a different commit, it has nothing to do with what you explain in your commit message.

Also the first rule with `position: fixed` does not apply to a children selector, it is confusing to find it here after children rules IMO. It should be at the with the base rules.

Reply:
@qsm-odoo ok done